src/modules/charging-import/lib/queries/charging.js

```diff
@@ -132,7 +132,73 @@ on conflict (external_id) do update set
   date_updated=NOW();
 `;
 
+const importChargeVersions = `INSERT INTO water.charge_versions (licence_ref, scheme, external_id, version_number, start_date, status, apportionment,
+error, end_date, billed_upto_date, region_code, date_created, date_updated, source, invoice_account_id, company_id, licence_id)
+SELECT
+  l."LIC_NO" AS licence_ref,
+  'alcs' AS scheme,
+  concat_ws(':', v."FGAC_REGION_CODE", v."AABL_ID", v."VERS_NO") as external_id,
+  v."VERS_NO"::integer AS version_number,
+  to_date(v."EFF_ST_DATE", 'DD/MM/YYYY') AS start_date,
+(CASE v."STATUS"
+  WHEN 'SUPER' THEN 'superseded'
+  WHEN 'DRAFT' THEN 'draft'
+  WHEN 'CURR' THEN 'current'
+END)::water.charge_version_status AS status,
+CASE v."APPORTIONMENT"
+  WHEN 'Y' THEN true
+  WHEN 'N' THEN false
+END AS apportionment,
+CASE v."IN_ERROR_STATUS"
+  WHEN 'Y' THEN true
+  WHEN 'N' THEN false
+END AS error,
+case v."EFF_END_DATE"
+  when 'null' then null
+  else to_date(v."EFF_END_DATE", 'DD/MM/YYYY')
+end AS end_date,
+case v."BILLED_UPTO_DATE"
+  when 'null' then null
+  else to_date(v."BILLED_UPTO_DATE", 'DD/MM/YYYY')
+end AS billed_upto_date,
+v."FGAC_REGION_CODE"::integer AS region,
+NOW() AS date_created,
+NOW() AS date_updated,
+'nald' AS source,
+ia.invoice_account_id,
+c.company_id,
+wl.licence_id
+FROM import."NALD_CHG_VERSIONS" v
```

Comment:
I think because `import."NALD_CHG_VERSIONS"` is the base table we will miss the gaps.  I wonder if we need to swap the order to:

```
from water_import.charge_versions_metadata cvm
left join import."NALD_CHG_VERSIONS" v on cvm.external_id = concat_ws(':', v."FGAC_REGION_CODE", v."AABL_ID", v."VERS_NO")
```

We will also need the change reason ID for the NALD gaps, so may need to add:
```
join water.change_reasons cr on cr.change_reason='NALD gap'
```

Reply:
> join water.change_reasons cr on cr.change_reason='NALD gap'

I'm sorry I forgot about that. Thanks for the check/reminder!